cmd/xl-storage.go

```diff
@@ -1153,6 +1153,8 @@ func (s *xlStorage) deleteVersions(ctx context.Context, volume, path string, fis
 					return err
 				}
 			}
+
+			resetDeadline <- struct{}{}
```

Comment:
We can send on `resetDeadline` channel inside a select statement with a default case to prevent `deleteVersions` from being blocked.